subprojects/core/src/main/java/org/gradle/internal/build/DefaultBuildWorkGraph.java

```diff
@@ -78,10 +78,8 @@ public void schedule(Collection<ExportedTaskNode> taskNodes) {
     }
 
     @Override
-    public void prepareForExecution() {
-        if (tasksScheduled) {
-            taskGraph.populate();
-        }
+    public void prepareForExecution(boolean alwaysPopulateWorkGraph) {
```

Comment:
We seem to need `alwaysPopulateWorkGraph`, since `taskScheduled` is false when loading from the configuration cache. I wonder if we could easily change that.